homeassistant/components/gardena_bluetooth/number.py

```diff
@@ -0,0 +1,148 @@
+
+
+@dataclass
+class GardenaBluetoothNumberEntityDescription(NumberEntityDescription):
+    """Description of entity."""
+
+    char: CharacteristicInt | CharacteristicUInt16 | CharacteristicLong = field(
+        default_factory=lambda: CharacteristicInt("")
+    )
+
+
+DESCRIPTIONS = (
+    GardenaBluetoothNumberEntityDescription(
+        key=Valve.manual_watering_time.uuid,
+        translation_key="manual_watering_time",
+        native_unit_of_measurement="s",
+        mode=NumberMode.BOX,
+        native_min_value=0.0,
+        native_max_value=24 * 60 * 60,
+        native_step=60,
+        entity_category=EntityCategory.CONFIG,
+        char=Valve.manual_watering_time,
+    ),
+    GardenaBluetoothNumberEntityDescription(
+        key=Valve.remaining_open_time.uuid,
+        translation_key="remaining_open_time",
+        native_unit_of_measurement="s",
+        native_min_value=0.0,
+        native_max_value=24 * 60 * 60,
+        native_step=60.0,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        char=Valve.remaining_open_time,
+    ),
+    GardenaBluetoothNumberEntityDescription(
+        key=DeviceConfiguration.rain_pause.uuid,
+        translation_key="rain_pause",
+        native_unit_of_measurement="d",
+        mode=NumberMode.BOX,
+        native_min_value=0.0,
+        native_max_value=127.0,
+        native_step=1.0,
+        entity_category=EntityCategory.CONFIG,
+        char=DeviceConfiguration.rain_pause,
+    ),
+    GardenaBluetoothNumberEntityDescription(
+        key=DeviceConfiguration.season_pause.uuid,
+        translation_key="season_pause",
+        native_unit_of_measurement="d",
+        mode=NumberMode.BOX,
+        native_min_value=0.0,
+        native_max_value=365.0,
+        native_step=1.0,
+        entity_category=EntityCategory.CONFIG,
+        char=DeviceConfiguration.season_pause,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up entity based on a config entry."""
+    coordinator: Coordinator = hass.data[DOMAIN][entry.entry_id]
+    entities: list[NumberEntity] = [
+        GardenaBluetoothNumber(coordinator, description)
+        for description in DESCRIPTIONS
+        if description.key in coordinator.characteristics
+    ]
+    entities.append(GardenaBluetoothRemainingOpenSetNumber(coordinator))
+    async_add_entities(entities)
+
+
+class GardenaBluetoothNumber(GardenaBluetoothEntity, NumberEntity):
+    """Representation of a number."""
+
+    entity_description: GardenaBluetoothNumberEntityDescription
+
+    def __init__(
+        self,
+        coordinator: Coordinator,
+        description: GardenaBluetoothNumberEntityDescription,
+    ) -> None:
+        """Initialize the Switchbot sensor."""
+        super().__init__(coordinator, {description.key})
+        self._attr_unique_id = f"{coordinator.address}-{description.key}"
+        self._attr_native_value = None
```

Comment:
Why set to `None` at startup?

Reply:
I thought this was undefined with fallback to method in base class like many other attr properties. Then it needs to be defined at init. I will check base class a bit later though.

Reply:
```suggestion
```